### PR TITLE
Fix Date picker problems caused by different time zones and DST.

### PR DESCRIPTION
**Pull request by hisapy**
Due to different time zones, the Date picker presented inconsistencies in formatting
and also displaying of days of the month and days of weeks. For example, in some time zones,
when due to DST the time forwards one hour at midnight, the browser simply skips the 00:00
hours. For example, in Paraguay where the time is forwarded 1 hour on Oct 1st,
new Date(2017, 9, 1, 0) returns Sun Oct 01 2017 01:00:00 GMT-0300 (-03).
This was causing problems such as rendering 2017-11-1 on thursday and not on wednesday, and
also broke the active and today displaying in the calendar.

To fix this issues, this commit does the following:

- Add required string prop timeZone to DatePickerContainer which defaults to UTC.
- Propagates timeZone prop to child/descendant components such as DatePicker, CalendarHeader, DatePickerCalendar, CalendarMonth and CalendarDate
- Always strips time when calculating the next date in CalendarMonth to avoid not displaying active/today dates.
- Updates Date Pickers docs' Formatting section.

## Screenshots of the problems taken from the docs page

1. React console showing the date assigned to the component

react-console

2. Component opened first time (wrong date and wrong days of the week)

dialog_opened_1st_time

3. Date picker switched to October. No active or today working; the screenshot was taken on October 30th.

dialog_oct

4. Date picker switched back to November. All dates displaced one day.

dialog_nov_after_oct

## Caveats

1. Could not make the new timeZone prop to appear in the PROP TYPES section of the docs. How can I do that?
2. This does not address Time picker issues.

### CODE REVIEW

File: src/js/Pickers/DatePickerContainer.js [hunk 1]
```diff
@@ -294,6 +294,12 @@ export default class DatePickerContainer extends PureComponent {
     }),
 
     /**
+     * The timeZone to be used in all formatting operations.
+     * For a full list of possible timeZone values check https://www.iana.org/time-zones.
+     */
+    timeZone: PropTypes.string.isRequired,
```
[the remark this answers is not in the document]
Comment (mlaursen) [reply src/js/Pickers/DatePickerContainer.js]: This is something that I need to make a better flow for.. You'd need to run `docgen` or the `docgen:create` commands again from the `/docs` folder to see those updates. I currently don't have hot-reloading set up for server code.

**Comment by mlaursen** [comment]
This looks good to me and thanks for updating a few of my tests to use enzyme! I've been meaning to migrate more of them over but never get the time.

File: src/js/Pickers/DatePickerContainer.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (hisapy) [reply src/js/Pickers/DatePickerContainer.js]: I think I tried all possible commands/scripts from the root dir and docs dir but I'm not sure. I'll take this into account for the next time :)

**Comment by johc** [comment]
Hi,
I noticed that the day of the week headers are incorrect in the calendar - it's shifted forward by a day. My timezone is GMT+13.

image

**Comment by hisapy** [comment]
We're working on this error in https://github.com/mlaursen/react-md/issues/634 ... thanks for the screenshot and your timezone info

**Pull request merged**